src/gnark_backend_glue/mod.rs

```diff
@@ -1,9 +1,97 @@
+use std::ffi::{CStr, CString};
+use std::os::raw::{c_char, c_uchar, c_void};
+use std::result;
+
 use acvm::{acir::circuit::Circuit, FieldElement};
+use anyhow::Result;
+use serde::Serialize;
+
+use self::acir_to_r1cs::RawR1CS;
+
+extern "C" {
+    fn Verify() -> c_uchar;
+    fn Prove(rawr1cs: GoString) -> *mut c_void;
+}
+
+#[derive(Debug)]
+#[repr(C)]
+struct GoString {
+    a: *const c_char,
+    b: i64,
```

Comment:
What are `a` and `b`?

Reply:
a = ptr to the first character of the string
b = length
I made it explicit in the struct now